modules/zookeeper/src/main/java/org/apache/ignite/spi/discovery/zk/ZookeeperDiscoverySpi.java

```diff
@@ -524,6 +504,14 @@ public Map<String, Object> getLocNodeAttrs() {
         return locNodeAttrs;
     }
 
+    /**
+     * Callback before join topology.
+     * @param locNode Local node.
+     */
+    public void beforeJoinTopology(ClusterNode locNode) {
```

Comment:
Do you think it is a good idea to specify in a javadoc that this method is intended for testing purposes only?